### PR TITLE
add eligibility check and protect export for stage and local

### PR DESCRIPTION
**Pull request by mchoun**
### Description

This PR is to add the `isEligible()` method to the `ThreeDomainSecureClient`. This method will allow the merchant to make a request to determine if the any payer action is required for 3DS authentication. If a payer action is required, it will obtain the authentication URL and make it available to the `show()` method for the buyer to authenticate. 

### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues

JIRA Ticket - DTPPCPSDK-2660

### Groups who should review (if applicable)

❤️ Thank you!

### CODE REVIEW

File: src/lib/security.js [hunk 1]
```diff
@@ -28,3 +29,13 @@ export function allowIframe(): boolean {
 export const protectedExport = (unprotectedExport) =>
   isPayPalDomain() ? unprotectedExport : undefined;
 /* eslint-enable no-confusing-arrow */
+
+// $FlowIssue
+export const localOrStageExport = (unprotectedExport) => {
```
Comment (ravishekhar) [inline src/lib/security.js:34]: Can we rename it to `devEnvOnlyExport` 🤔 ?

File: src/three-domain-secure/component.jsx [hunk 2]
```diff
@@ -23,29 +86,81 @@ const parseSdkConfig = ({ sdkConfig, logger }): SdkConfig => {
...
-    return new ZalgoPromise((resolve) => {
-      resolve(false);
-    });
+  async isEligible(merchantPayload: MerchantPayloadData): Promise<boolean> {
+    const data = parseMerchantPayload({ merchantPayload });
+    try {
+      // $FlowFixMe
+      const { status, links } = await this.request<requestData, responseBody>({
+        method: "POST",
+        url: `${this.sdkConfig.paypalApiDomain}/v2/payments/payment`,
```
Comment (ravishekhar) [inline src/three-domain-secure/component.jsx:146]: I feel we should keep all the endpoints in a single const file.

**Review by ravishekhar — APPROVED** [review]
Left a few trivial comments. Looks great otherwise.

File: src/three-domain-secure/component.jsx (hunk 2 above)
Comment (mchoun) [reply src/three-domain-secure/component.jsx]: Done!

File: src/lib/security.js (hunk 1 above)
Comment (mchoun) [reply src/lib/security.js]: Much better name :)

**Review by Dlbandy04 — APPROVED** [review]

**Pull request merged**